### PR TITLE
Add consul impl

### PR DESCRIPTION
**Pull request by yairogen**

### CODE REVIEW

File: master-slave/src/main/java/com/cisco/oss/foundation/cluster/masterslave/consul/ConsulMastershipElector.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                }
+            }
+        },"Infinite-Reconnect").start();
+    }
+
+    @Override
+    public boolean isReady() {
+        try {
+            if(consul != null){
+                consul.agentClient().ping();
```
Comment (igreenfield) [inline master-slave/src/main/java/com/cisco/oss/foundation/cluster/masterslave/consul/ConsulMastershipElector.java]: Does it return something?

File: master-slave/pom.xml [hunk 2]
```diff
@@ -215,6 +215,23 @@
             <artifactId>de.flapdoodle.embed.mongo</artifactId>
             <version>1.50.2</version>
```
Comment (igreenfield) [inline master-slave/pom.xml]: is it in test scope?

Comment (yairogen) [inline master-slave/pom.xml]: changed to test scope

File: master-slave/src/main/java/com/cisco/oss/foundation/cluster/masterslave/consul/ConsulMastershipElector.java (hunk 1 above)
Comment (yairogen) [inline master-slave/src/main/java/com/cisco/oss/foundation/cluster/masterslave/consul/ConsulMastershipElector.java]: no it doesn't. only failure indicates ping didn't work.

**Pull request merged**